nvfuser/__init__.py

```diff
@@ -50,6 +51,54 @@ def disable_automatic_serialization():
     atexit.unregister(_C.serialize)
 
 
+class DistributedTensor(torch.Tensor):
```

Comment:
I felt the naming here could be confusing. Since it's a DistributedTensor that behaves as a local tensor in computation.

But I *think* the idea is for experiments and we'll likely converge to some other upstream/framework dtensor representation later, I'm fine with this as-is.

Reply:
Good point -- I'll think about it. That brings an alternative to let DistributedTensor have the same metadata as the **global** tensor while still owning a **local** tensor. DTensor and XLAShardedTensor follow that convention. This way, the global shape can be retrieved using `DistributedTensor.shape`. The biggest benefit, I think, will be to support uneven split where `local_size * mesh_size` doesn't necessarily equal `global_size`. It will make tests harder to write because they'll have to access the local tensor by explicitly calling `DistributedTensor.local`.